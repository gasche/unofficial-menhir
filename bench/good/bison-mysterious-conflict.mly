/* This grammar is not LALR(1). It is taken from the Bison manual. */

%token ID COMMA COLON
%start <unit> def

%%

def:
  param_spec return_spec COMMA
    { () }

param_spec:
  typ
    { () }
| separated_nonempty_list(COMMA, name) COLON typ
    { () }

return_spec:
  typ
    { () }
| name COLON typ
    { () }

typ:
  ID
    { () }

name:
  ID
    { () }

