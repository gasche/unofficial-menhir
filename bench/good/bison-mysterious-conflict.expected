%start def
%token ID
%token COMMA
%token COLON

%type <unit> def
%%

def:
| param_spec return_spec COMMA
    { () }

param_spec:
| typ
    { () }
| separated_nonempty_list_COMMA_name_ COLON typ
    { () }

return_spec:
| typ
    { () }
| name COLON typ
    { () }

typ:
| ID
    { () }

name:
| ID
    { () }

separated_nonempty_list_COMMA_name_:
| x = name
    { [ x ] }
| x = name COMMA xs = separated_nonempty_list_COMMA_name_
    { x :: xs }

%%



